src/net.cpp

```diff
@@ -611,6 +611,8 @@ void CNode::copyStats(CNodeStats &stats, const std::vector<bool> &m_asmap)
     stats.addrLocal = addrLocalUnlocked.IsValid() ? addrLocalUnlocked.ToString() : "";
 
     X(m_conn_type);
+
+    stats.addr_relay_enabled = RelayAddrsWithConn();
```

Comment:
This isn't thread safe. By accessing `m_addr_known` from outside the message handler thread (eg an rpc worker thread or qt), this opens race conditions where `m_addr_known` is being set by the message handler thread and read by an rpc thread concurrently. In such cases, the thread reading the value could see the old value, the new value or garbage.

Reply:
ayo, good catch. it feels like overkill to introduce a lock just to expose as a boolean, so I've dropped this commit for now. will some of your addr-in-net-processing refactor will make it so we can do this safely?

Reply:
> will some of your addr-in-net-processing refactor will make it so we can do this safely?

Hopefully, yes!